Bump Ruby to 2.6.6 and use GitHub Actions instead of Jenkins

issyl0 opened the pull request:
- This moves us ever closer towards all the easy migrations earmarked in [RFC 123](https://github.com/alphagov/govuk-rfcs/blob/master/rfc-123-github-actions-ci.md) being done. 😱
- This also adds RuboCop and the corresponding `lint` Rake task to the `bundle exec rake` defaults. And the version of Ruby was behind the latest.

https://trello.com/c/PzpcnUdS/144-migrate-all-skipdeploytointegration-repos-ci-to-github-actions

issyl0 committed:
Bump Ruby to 2.6.6

issyl0 committed:
ci: Use GitHub Actions instead of Jenkins

issyl0 committed:
Add RuboCop and make it run as part of `bundle exec rake`

- All our other repos have it, so why not this one.

issyl0 committed:
Autocorrect all possible RuboCop offenses

benthorner commented on spec/spec_helper.rb:
```diff
@@ -47,57 +47,55 @@
-  # as the one that triggered the failure.
-  Kernel.srand config.seed
-=end
+  # The settings below are suggested to provide a good initial experience
```
The auto-correct has messed up these comments. We should just delete them.

benthorner commented on .rubocop.yml:
```diff
@@ -1,3 +1,8 @@
   rubocop-govuk:
     - config/default.yml
+
+Metrics/BlockLength:
```
Maybe this can wait for https://github.com/alphagov/rubocop-govuk/pull/65?

issyl0 replied:
Cool. I await a new release!

issyl0 committed:
spec_helper: Delete superfluous comments

issyl0 commented:
In accordance with the still draft but broadly sensible [Actions conventions doc](https://github.com/alphagov/govuk-developer-docs/pull/2533), I renamed this file `ci.yml` and added a name.
